### ISSUE TITLE
Qt `6.8.0` can cause stuck mouse hover states (`pauseOnHoverDuration`)

### ISSUE THREAD

**Issue by brian6932** [Chatterino/chatterino2#5704]
### Checklist

- [X] I'm reporting a problem with Chatterino
- [X] I've verified that I'm running **the most** recent nightly build or stable release
- [X] I've looked for my problem on the [wiki](https://wiki.chatterino.com/Help/)
- [X] I've searched the [issues and pull requests](https://github.com/Chatterino/chatterino2/issues?q=) for similar looking reports

### Describe your issue

~~`pauseOnHoverDuration` was never really consistent, in the sense that sometimes it wouldn't show the icon, despite the cursor being on a split, it would still pause. However~~ < Qt `6.8.0` doesn't get into a stuck _paused_ state. I believe a minimum split size's required to reproduce this, so I included the relevant `/settings/*.json` files.

<details>
	<summary>settings.json</summary>

```json
{"misc":{"currentVersion":"2.5.1","lockNotebookLayout":false,"askOnTabVisibilityToggle":false},"behaviour":{"autorun":false,"pauseOnHoverDuration":-1.0},"hotkeys":{"addedDefaults":["change channel","close popup window","create clip","delete","emote picker","focus down","focus left","focus right","focus up","go to end of input","go to end of input with selection","go to start of input","go to start of input with selection","new popup window","new popup window from tab","new split","new tab","next message","open debug popup","open overlay","open quick switcher","open settings","popup accept","popup focus search box","popup reject","popup scroll down","popup scroll up","popup select last tab","popup select next tab","popup select previous tab","popup select tab #1","popup select tab #2","popup select tab #3","popup select tab #4","popup select tab #5","popup select tab #6","popup select tab #7","popup select tab #8","previous message","reconnect","reload emotes","remove tab","reopen split","scroll page down","scroll page up","scroll to bottom","scroll to top","select last tab","select next tab","select previous tab","select tab #1","select tab #2","select tab #3","select tab #4","select tab #5","select tab #6","select tab #7","select tab #8","send message","send message and keep text","show global search","show search","toggle local r9k","toggle overlay click-through","toggle tab visibility","zoom in","zoom out","zoom reset"],"close popup window":{"action":"delete","keySequence":"Esc","category":"popupWindow","arguments":[]},"popup accept":{"action":"accept","keySequence":"Return","category":"popupWindow","arguments":[]},"popup focus search box":{"action":"search","keySequence":"Ctrl+F","category":"popupWindow","arguments":[]},"popup reject":{"action":"reject","keySequence":"Esc","category":"popupWindow","arguments":[]},"popup scroll down":{"action":"scrollPage","keySequence":"PgDown","category":"popupWindow","arguments":["down"]},"popup scroll up":{"action":"scrollPage","keySequence":"PgUp","category":"popupWindow","arguments":["up"]},"popup select last tab":{"action":"openTab","keySequence":"Ctrl+9","category":"popupWindow","arguments":["last"]},"popup select next tab":{"action":"openTab","keySequence":"Ctrl+Tab","category":"popupWindow","arguments":["next"]},"popup select previous tab":{"action":"openTab","keySequence":"Ctrl+Shift+Tab","category":"popupWindow","arguments":["previous"]},"popup select tab #1":{"action":"openTab","keySequence":"Ctrl+1","category":"popupWindow","arguments":["0"]},"popup select tab #2":{"action":"openTab","keySequence":"Ctrl+2","category":"popupWindow","arguments":["1"]},"popup select tab #3":{"action":"openTab","keySequence":"Ctrl+3","category":"popupWindow","arguments":["2"]},"popup select tab #4":{"action":"openTab","keySequence":"Ctrl+4","category":"popupWindow","arguments":["3"]},"popup select tab #5":{"action":"openTab","keySequence":"Ctrl+5","category":"popupWindow","arguments":["4"]},"popup select tab #6":{"action":"openTab","keySequence":"Ctrl+6","category":"popupWindow","arguments":["5"]},"popup select tab #7":{"action":"openTab","keySequence":"Ctrl+7","category":"popupWindow","arguments":["6"]},"popup select tab #8":{"action":"openTab","keySequence":"Ctrl+8","category":"popupWindow","arguments":["7"]},"change channel":{"action":"changeChannel","keySequence":"Ctrl+R","category":"split","arguments":[]},"create clip":{"action":"createClip","keySequence":"Alt+X","category":"split","arguments":[]},"delete":{"action":"delete","keySequence":"Ctrl+W","category":"split","arguments":[]},"focus down":{"action":"focus","keySequence":"Alt+Down","category":"split","arguments":["down"]},"focus left":{"action":"focus","keySequence":"Alt+Left","category":"split","arguments":["left"]},"focus right":{"action":"focus","keySequence":"Alt+Right","category":"split","arguments":["right"]},"focus up":{"action":"focus","keySequence":"Alt+Up","category":"split","arguments":["up"]},"open debug popup":{"action":"debug","keySequence":"F10","category":"split","arguments":[]},"open overlay":{"action":"popupOverlay","keySequence":"Ctrl+Alt+N","category":"split","arguments":[]},"reconnect":{"action":"reconnect","keySequence":"Ctrl+F5","category":"split","arguments":[]},"reload emotes":{"action":"reloadEmotes","keySequence":"F5","category":"split","arguments":[]},"scroll page down":{"action":"scrollPage","keySequence":"PgDown","category":"split","arguments":["down"]},"scroll page up":{"action":"scrollPage","keySequence":"PgUp","category":"split","arguments":["up"]},"scroll to bottom":{"action":"scrollToBottom","keySequence":"Ctrl+End","category":"split","arguments":[]},"scroll to top":{"action":"scrollToTop","keySequence":"Ctrl+Home","category":"split","arguments":[]},"show global search":{"action":"showGlobalSearch","keySequence":"Ctrl+Shift+F","category":"split","arguments":[]},"show search":{"action":"showSearch","keySequence":"Ctrl+F","category":"split","arguments":[]},"toggle overlay click-through":{"action":"toggleOverlayInertia","keySequence":"Ctrl+Shift+U","category":"split","arguments":["all"]},"emote picker":{"action":"openEmotesPopup","keySequence":"Ctrl+E","category":"splitInput","arguments":[]},"go to end of input":{"action":"cursorToEnd","keySequence":"End","category":"splitInput","arguments":["withoutSelection"]},"go to end of input with selection":{"action":"cursorToEnd","keySequence":"Shift+End","category":"splitInput","arguments":["withSelection"]},"go to start of input":{"action":"cursorToStart","keySequence":"Home","category":"splitInput","arguments":["withoutSelection"]},"go to start of input with selection":{"action":"cursorToStart","keySequence":"Shift+Home","category":"splitInput","arguments":["withSelection"]},"next message":{"action":"nextMessage","keySequence":"Down","category":"splitInput","arguments":[]},"previous message":{"action":"previousMessage","keySequence":"Up","category":"splitInput","arguments":[]},"send message":{"action":"sendMessage","keySequence":"Return","category":"splitInput","arguments":[]},"send message and keep text":{"action":"sendMessage","keySequence":"Ctrl+Return","category":"splitInput","arguments":["keepInput"]},"new popup window":{"action":"popup","keySequence":"Ctrl+N","category":"window","arguments":["split"]},"new popup window from tab":{"action":"popup","keySequence":"Ctrl+Shift+N","category":"window","arguments":["window"]},"new split":{"action":"newSplit","keySequence":"Ctrl+T","category":"window","arguments":[]},"new tab":{"action":"newTab","keySequence":"Ctrl+Shift+T","category":"window","arguments":[]},"open quick switcher":{"action":"openQuickSwitcher","keySequence":"Ctrl+K","category":"window","arguments":[]},"open settings":{"action":"openSettings","keySequence":"Ctrl+P","category":"window","arguments":[]},"remove tab":{"action":"removeTab","keySequence":"Ctrl+Shift+W","category":"window","arguments":[]},"reopen split":{"action":"reopenSplit","keySequence":"Ctrl+G","category":"window","arguments":[]},"select last tab":{"action":"openTab","keySequence":"Ctrl+9","category":"window","arguments":["last"]},"select next tab":{"action":"openTab","keySequence":"Ctrl+Tab","category":"window","arguments":["next"]},"select previous tab":{"action":"openTab","keySequence":"Ctrl+Shift+Tab","category":"window","arguments":["previous"]},"select tab #1":{"action":"openTab","keySequence":"Ctrl+1","category":"window","arguments":["0"]},"select tab #2":{"action":"openTab","keySequence":"Ctrl+2","category":"window","arguments":["1"]},"select tab #3":{"action":"openTab","keySequence":"Ctrl+3","category":"window","arguments":["2"]},"select tab #4":{"action":"openTab","keySequence":"Ctrl+4","category":"window","arguments":["3"]},"select tab #5":{"action":"openTab","keySequence":"Ctrl+5","category":"window","arguments":["4"]},"select tab #6":{"action":"openTab","keySequence":"Ctrl+6","category":"window","arguments":["5"]},"select tab #7":{"action":"openTab","keySequence":"Ctrl+7","category":"window","arguments":["6"]},"select tab #8":{"action":"openTab","keySequence":"Ctrl+8","category":"window","arguments":["7"]},"toggle local r9k":{"action":"toggleLocalR9K","keySequence":"Ctrl+H","category":"window","arguments":[]},"toggle tab visibility":{"action":"setTabVisibility","keySequence":"Ctrl+U","category":"window","arguments":["toggle"]},"zoom in":{"action":"zoom","keySequence":"Ctrl++","category":"window","arguments":["in"]},"zoom out":{"action":"zoom","keySequence":"Ctrl+-","category":"window","arguments":["out"]},"zoom reset":{"action":"zoom","keySequence":"Ctrl+0","category":"window","arguments":["reset"]}},"appearance":{"showEmptyInputBox":false,"windowAlwaysOnTop":true,"tabVisibility":0}}
```
</details>

<details>
	<summary>window-layout.json</summary>

```json
{"windows":[{"emotePopup":{"height":0,"width":0,"x":0,"y":0},"height":1081,"tabs":[{"highlightsEnabled":true,"selected":true,"splits2":{"flexh":1,"flexv":1,"items":[{"flexh":1,"flexv":0.5971014492753624,"items":[{"data":{"name":"tmiloadtesting3","type":"twitch"},"filters":[],"flexh":1,"flexv":1,"moderationMode":false,"type":"split"},{"data":{"name":"twitchmedia_qs_10","type":"twitch"},"filters":[],"flexh":1,"flexv":1,"moderationMode":false,"type":"split"}],"type":"horizontal"},{"data":{"name":"","type":"twitch"},"filters":[],"flexh":1,"flexv":1.4028985507246376,"moderationMode":false,"type":"split"}],"type":"vertical"}}],"type":"main","width":598,"x":956,"y":114}]}
```
</details>

### Screenshots

<details>
	<summary>6.7.2</summary>

![](https://github.com/user-attachments/assets/529083ad-b48c-4cc1-9752-f5896bd1758d)
</details>

<details>
	<summary>6.8.0</summary>

![](https://github.com/user-attachments/assets/2ed41a83-68f9-431c-981c-b5272e815a3e)
</details>

### OS and Chatterino Version

68304272a26c12b7d72fa3d95a732471341e3d15 built on 2024-11-04 with Qt 6.8.0, MSVC 194134123 Running on Windows 10 Version 21H2, kernel: 10.0.19044

**Comment by Nerixyz** [comment]
This only happens because the input box is "hidden". Reduced to

```cpp
#include <QApplication>
#include <QWidget>
#include <cassert>

int main(int argc, char **argv) {
  QApplication app(argc, argv);

  QWidget parent(nullptr, Qt::FramelessWindowHint);
  parent.resize(200, 200);

  QWidget *emptyChild = new QWidget(&parent);
  emptyChild->setGeometry(0, 100, 200, 0);

  assert(parent.childAt(100, 100) == nullptr);
}
```

Filed [QTBUG-131001](https://bugreports.qt.io/browse/QTBUG-131001).

**Comment by pajlada** [comment]
@brian6932 could you check if this is fixed for you on latest nightly?

**Comment by Nerixyz** [comment]
> `pauseOnHoverDuration` was never really consistent, in the sense that sometimes it wouldn't show the icon, despite the cursor being on a split, it would still pause. 

Yea my PR only fixed this part. The other one is a Qt bug.

**Comment by Nerixyz** [comment]
Fixed in Qt 6.8.1.

**Comment by brian6932** [comment]
Can no longer reproduce on 6.8.1. Thank you Tom Fors, very cool.
